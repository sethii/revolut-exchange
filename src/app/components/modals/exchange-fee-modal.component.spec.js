import React from 'react';
import renderer from 'react-test-renderer';
import { toClass } from 'recompose';
import { mount } from 'enzyme';

import { ExchangeFeeModal } from './exchange-fee-modal.component';

describe('Exchange fee modal component', () => {
  it('renders exchange fee modal', () => {
    const ExchangeFeeModalComponent = toClass(ExchangeFeeModal);

    expect(renderer.create(<ExchangeFeeModalComponent display={true} closeModal={jest.fn()}/>).toJSON()).toMatchSnapshot();
  });

  it('renders null when display false passed', () => {
    const ExchangeFeeModalComponent = toClass(ExchangeFeeModal);

    expect(renderer.create(<ExchangeFeeModalComponent display={false} closeModal={jest.fn()}/>).toJSON()).toMatchSnapshot();
  });

  it('handles modal close click', () => {
    const ExchangeFeeModalComponent = toClass(ExchangeFeeModal);
    const closeModal = jest.fn();

    const wrapper = mount(<ExchangeFeeModalComponent display={true} closeModal={closeModal}/>);

    wrapper.find('a').simulate('click');

    expect(closeModal).toBeCalled();
  });
});
