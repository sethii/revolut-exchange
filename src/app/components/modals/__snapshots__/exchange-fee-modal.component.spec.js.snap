// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exchange fee modal component renders exchange fee modal 1`] = `
.glamor-5 {
  border-radius: 10px;
  width: 500px;
  height: auto;
  padding: 20px;
  background-color: white;
  margin: 0 auto;
  font-size: 14px;
}

.glamor-1 {
  font-size: 24px;
}

.glamor-3 {
  font-size: 16px;
  color: #4169E1;
}

.glamor-3:hover {
  cursor: pointer;
}

<div
  className="glamor-4 glamor-5"
>
  <h1
    className="glamor-0 glamor-1"
  >
    FX exchange limit fee
  </h1>
  <p>
    You've reached FX rate limit of $6000 per month. Future exchange will proceed with 0,5% fee.
  </p>
  <a
    className="glamor-2 glamor-3"
    onClick={[Function]}
  >
    GOT IT
  </a>
</div>
`;

exports[`Exchange fee modal component renders null when display false passed 1`] = `null`;
